### PR TITLE
chat2 improvements

### PR DESCRIPTION
**Pull request by jm-clius**
This is an incremental PR towards #399

It adds the following improvements to the `chat2` application:
1. Logs are now written to a file (`chat2.log`) rather than to `stdout`. The latter is now only used for the chat app itself.
2. Added a chat prompt (`>> `).
3. A nickname can now be set before starting a chat session and changed using the `/nick` command. This is prepended to each published message.
4. Each message in the chat is now prepended by a UTC timestamp.
5. If no static node is set, one will now be selected (at random) from the list of `wakuv2.test` nodes published at `https://fleets.status.im`. (In order to test this, simply run `./build/chat2` with no `--staticnode` argument. `--storenode` unfortunately still needs to be configured - I ran into some issues while trying to automate its selection.)

### Further improvements:
1. Store node selection in similar manner to static node selection.
2. Instruction manual and better configuration
??

This is very much a WIP, so it would be great to get your initial feedback after trying the app :D

### CODE REVIEW

**Review by staheri14 — APPROVED** [review]
Great improvement on chat2!

File: examples/v2/chat2.nim [hunk 1]
```diff
@@ -70,6 +71,23 @@ proc connectToNodes(c: Chat, nodes: seq[string]) {.async.} =
   await c.node.connectToNodes(nodes)
   c.connected = true
 
+proc selectRandomNode(): string =
+  randomize()
```
Comment (oskarth) [inline examples/v2/chat2.nim:75]: I don't think it really matters, but I think in Nim `rng` is preferred a la https://github.com/status-im/nim-eth/blob/4e58eb48ce1b1a06d28802ad22215de9ff92a916/tests/p2p/test_routing_table.nim#L7-L18

**Review by oskarth — APPROVED** [review]
Awesome stuff, looks great! The only thing is that it doesn't seem to display historical messages for me anymore.

> cat chat2.log  | grep potato also returns zero potatos, I would expect to find at least one if it accidentally ended up in that stream

**Comment by oskarth** [comment]
Update: I saw historical messages from this branch on master with previous hardcoded node. Now it randomly connects to 

```
Welcome, oskarth!
No static peers configured. Choosing one at random from test fleet...
Connecting to /ip4/134.209.139.210/tcp/30303/p2p/16Uiu2HAmPLe7Mzm8TsYUubgCAW1aJoeFScxrLj8ppHFivPo97bUZ
Connecting to nodes
Listening on
 /ip4/0.0.0.0/tcp/60000/p2p/16Uiu2HAmMxwff3HHW9sYZ71sSMCAGfW7v28Gs846aQv9Vpd8BZvM
```

If I call it with

```
./build/chat2 --storenode:/ip4/134.209.139.210/tcp/30303/p2p/16Uiu2HAmPLe7Mzm8TsYUubgCAW1aJoeFScxrLj8ppHFivPo97bUZ --staticnode:/ip4/134.209.139.210/tcp/30303/p2p/16Uiu2HAmPLe7Mzm8TsYUubgCAW1aJoeFScxrLj8ppHFivPo97bUZ
```

I do see historical messages, but not the ones I previously sent. If I remove the staticnode argument and leave storenode argument I don't.

tldr seems like cluster connection issue, and some gotcha when only one of arguments is supplied?

**Comment by oskarth** [comment]
`metrics error: The number of label values doesn't match the number of labels.` is redirected to STDOUT not error log stream as well

File: examples/v2/chat2.nim [hunk 2]
```diff
@@ -139,7 +166,10 @@ proc writeAndPrint(c: Chat) {.async.} =
     else:
       # XXX connected state problematic
       if c.started:
-        c.publish(line)
+        # Get message timestamp
+        let time = getTime().utc().format("'<'HH:mm'>'")
```
[the remark this answers is not in the document]
Comment (jm-clius) [reply examples/v2/chat2.nim:170]: Mmm. I agree, but was worried that it might clutter the chat space if every message is prepended by a timestamp, date and nick. One idea is to have some kind of "day" divider between different dates, but that may require more sophisticated state-keeping than is necessary in a simple chat app. I might just try prepending the shortest date format possible to every message and see how it affects UX 😄, e.g.
`<Mar 3, 7:15> nick: this is a chat message`

**Comment by jm-clius** [comment]
> I do see historical messages, but not the ones I previously sent. If I remove the staticnode argument and leave storenode argument I don't.
> 
> tldr seems like cluster connection issue, and some gotcha when only one of arguments is supplied?

Yes, there's something weird going on with the cluster, likely a connection issue. In fact, if I connect to `/ip4/104.154.239.128/tcp/30303/p2p/16Uiu2HAmPLe7Mzm8TsYUubgCAW1aJoeFScxrLj8ppHFivPo97bUZ` as `staticnode` (whether randomly selected or as explicit argument, and regardless of what `storenode` is configured) I see no historical messages and my chat messages aren't relayed to any of the other peers. The other two cluster nodes seem to work, though, either as `staticnode` or `storenode`.
